ion-schema/src/system.rs

```diff
@@ -1835,4 +1986,108 @@ mod schema_system_tests {
         assert_eq!(schema.open_content(), &expected_open_content);
         Ok(())
     }
+
+    #[test]
+    fn unexpected_fields_in_isl_v2_0_header() {
+        // map with (id, ion content)
+        let map_authority = [(
+            "sample.isl",
+            r#"
+                schema_header::{
+                    user_reserved_fields: {
+                        schema_header: [ foo, bar ],
+                        type: [ baz ]
+                    },
+                    baz: "this is an unexpected field in schema header"
+                }
+                
+                type::{
+                  name: my_type,
+                  type: string,
+                }
+                
+                schema_footer::{}
+            "#,
+        )];
+        let mut schema_system =
+            SchemaSystem::new(vec![Box::new(MapDocumentAuthority::new(map_authority))]);
+        let schema = schema_system.load_isl_schema_v2_0("sample.isl");
+        assert!(schema.is_err());
+    }
+
+    #[test]
+    fn unexpected_fields_in_isl_v2_0_footer() {
+        // map with (id, ion content)
+        let map_authority = [(
+            "sample.isl",
+            r#"
+                schema_header::{
+                    user_reserved_fields: {
+                        schema_footer: [ foo, bar ],
+                        sdhema_header: [ baz ]
+                    },
+                }
+                
+                type::{
+                  name: my_type,
+                  type: string,
+                }
+                
+                schema_footer::{
+                    baz: "this is an unexpected field in schema header"
+                }
```

Comment:
According to the spec, this is an ISL 1.0 schema document (since there is no `$ion_schema_2_0` marker), and so it should be valid to have all of these extra fields.

Reply:
Yes, That is correct. This should refactored with the change in `load_isl_schema` function that has version marker check.

> Oh yes! Thanks for catching this. I can probably add the check inside `load_isl_schema` for version marker one doesn't need to know what version it is using.